src/execution_plan/ops/op_delete.c

```diff
@@ -177,6 +178,23 @@ static Record DeleteConsume(OpBase *opBase) {
 			array_append(op->deleted_edges, *e);
 			// If evaluating the expression allocated any memory, free it.
 			SIValue_Free(value);
+		} else if(type & T_PATH) {
+			Path *p = (Path *)value.ptrval;
+
+			size_t nodeCount = Path_NodeCount(p);
+			size_t edgeCount = Path_Len(p);
+
+			for(size_t i = 0; i < nodeCount - 1 ; i++) {
+				Node *n = Path_GetNode(p, i);
+				array_append(op->deleted_nodes, *n);
+				Edge *e = Path_GetEdge(p, i);
+				array_append(op->deleted_edges, *e);
+			}
+			if(nodeCount > 0) {
+				Node *n = Path_GetNode(p, nodeCount - 1);
+				array_append(op->deleted_nodes, *n);
+			}
```

Comment:
Thanks. I rewrote the code in [b02a867](https://github.com/RedisGraph/RedisGraph/pull/2567/commits/b02a8673b2d87598822059512e7be4ac416f0107) based on your suggestion, but I did some changes because your code does not get the edge at position 0 and I noticed that the edgeCount variable was not being used.

Reply:
sorry and this one?
```
if(nodeCount > 0) {
  	Node *n = Path_GetNode(p, 0);
	array_append(op->deleted_nodes, *n);

	for(size_t i = 1; i < nodeCount; i++) {
		Node *n = Path_GetNode(p, i);
		array_append(op->deleted_nodes, *n);
		Edge *e = Path_GetEdge(p, I - 1);
		array_append(op->deleted_edges, *e);
	}
}
```

Reply:
Ok. It's clearer and works. Change done in [892bebb](https://github.com/RedisGraph/RedisGraph/pull/2567/commits/892bebba360a502c1dd3dd5ec66e1d1d0fa84f83). Thanks